### ISSUE TITLE
add option for scheduler step once per loop for multi-GPU scenery

### ISSUE THREAD

**Issue by Wongboo** [huggingface/accelerate#2165]
I find that when running 
`
scheduler.step()
`
in a loop of 
`
for batch in training_dataloader:
`
while cause `scheduler` step number of gpus per loop, is it possible to let it step only once per loop?
(I test it on single node, don't know multi-node scenery)

**Comment by SimonBenhamou** [comment]
I can confirm I observe the same behavior on my side. Is this a bug ?

**Comment by SimonBenhamou** [comment]
My current workaround is to not apply the prepare method on the scheduler anymore.

**Comment by muellerzr** [comment]
There is an argument in Accelerator to disable this (step with optimizer iirc)

**Comment by muellerzr** [comment]
As this isn’t a bug, it’s intentional
